mixer/cmd/build.go

```diff
@@ -199,113 +218,158 @@ var buildFormatNewCmd = &cobra.Command{
+		// <bundle>/ directories that are empty. When the manifest creation happens it will
+		// mark all files in that bundles as deleted.
+		f := func(filename string) error {
+			bdir := filepath.Join(b.Config.Builder.ServerStateDir, "image", b.MixVer, filename)
+			// This bundle is deprecated, remove the info file
+			if err = os.RemoveAll(bdir + "-info"); err != nil {
+				return err
+			}
+			// Create the empty dir for update to mark all files as deleted
+			if err = os.MkdirAll(bdir, 0755); err != nil {
+				return errors.Wrapf(err, "Failed to create bundle directory: %s", bdir)
+			}
+			return nil
+		}
+		fmt.Println("Removing deprecated bundle-info files...")
+		// Remove deleted bundles and replace with empty dirs for update to mark as deleted
+		if err = b.ModifyBundles(f); err != nil {
 			fail(err)
 		}
 
+		// Link the +20 bundles to the +10 so we are building the update with the same
+		// underlying content. The only things that might change are the manifests
+		// (potentially the pack and full-file formats as well, though this is very
+		// rare).
 		ver, err = strconv.Atoi(b.MixVer)
 		if err != nil {
 			fail(err)
 		}
+		if err = b.UpdateMixVer(ver - 10); err != nil {
+			failf("Couldn't update Mix Version")
+		}
+		source := filepath.Join(b.Config.Builder.ServerStateDir, "image", strconv.Itoa(ver))
+		dest := filepath.Join(b.Config.Builder.ServerStateDir, "image", b.MixVer)
+		fmt.Println(" Copying +20 bundles to +10 bundles")
+		if err = helpers.RunCommandSilent("cp", "-al", source, dest); err != nil {
+			failf("Failed to copy +10 bundles to +20: %s\n", err)
+		}
 
-		// Build the +20 update so we don't have to switch tooling in between
+		// Set the format back to old for the actual build update
+		b.State.Mix.Format = oldFormat
+
+		// Build the update content for the +10 build
 		params := builder.UpdateParameters{
-			MinVersion:    ver,
-			Format:        strconv.Itoa(newFormat),
+			MinVersion:    buildFlags.minVersion,
+			Format:        b.State.Mix.Format,
 			Publish:       !buildFlags.noPublish,
 			SkipSigning:   buildFlags.noSigning,
 			SkipFullfiles: buildFlags.skipFullfiles,
 			SkipPacks:     buildFlags.skipPacks,
 		}
-		err = b.BuildUpdate(params)
-		if err != nil {
+		if err = b.BuildUpdate(params); err != nil {
 			failf("Couldn't build update: %s", err)
 		}
 
-		// Copy +20 chroots to +10 so we can build last formatN build with the
-		// same content
-		prevVersion, err := strconv.Atoi(b.MixVer)
-		if err != nil {
-			fail(err)
-		}
-		prevVersion -= 10
-		source := filepath.Join(b.Config.Builder.ServerStateDir, "image", b.MixVer)
-		dest := filepath.Join(b.Config.Builder.ServerStateDir, "image", strconv.Itoa(prevVersion))
-		fmt.Println(" Copying +20 chroots to +10 chroots")
-		if err = helpers.RunCommandSilent("cp", "-al", source, dest); err != nil {
-			failf("Failed to copy +20 chroots to +10: %s\n", err)
+		zeroVer := strconv.Itoa(ver - 20)
```

Comment:
this could be set to `oldFormatVer - 10` if it was defined, but would probably be even easier to follow if you initiated it earlier on using the last version of the mix before the format bump started or something.